outbox/src/main/kotlin/com/ritense/outbox/OutboxMessageRepository.kt

```diff
@@ -31,7 +31,7 @@ interface OutboxMessageRepository : JpaRepository<OutboxMessage, UUID> {
      */
     @QueryHints(value = [QueryHint(name = JPA_LOCK_TIMEOUT, value = SKIP_LOCKED)])
     @Lock(PESSIMISTIC_WRITE)
-    fun findTopByOrderByCreatedOnAsc(): OutboxMessage?
+    fun findFirstBy(): OutboxMessage?
```

Comment:
What does `findFirstBy` create as a query? By what field does it find the first?

Reply:
The generated query will be:
``` SELECT * from outbox_message LIMIT 1 FOR UPDATE SKIP LOCKED```
There is no ordering anymore because that locked the entire MySql table. But the messages are no longer consumed in the correct order. Which is a big loss imo

Reply:
I'd like to see a separate repository for Postgres and Mysql to solve this. I can live with MySQL not having this feature, but Postgres should still have it.